gooddata-java/src/main/java/com/gooddata/sdk/service/hierarchicalconfig/HierarchicalConfigService.java

```diff
@@ -0,0 +1,121 @@
+import static com.gooddata.sdk.common.util.Validate.notNull;
+import static com.gooddata.sdk.model.hierarchicalconfig.SettingItem.PROJECT_SETTING_TEMPLATE;
+import static com.gooddata.sdk.model.hierarchicalconfig.Settings.PROJECT_SETTINGS_TEMPLATE;
+import static com.gooddata.sdk.service.hierarchicalconfig.HierarchicalConfigValidator.isItProjectSetting;
+
+/**
+ * Provides hierarchical configuration management. For more detailed description, see this documentation
+ * https://help.gooddata.com/api#/reference/hierarchical-configuration
+ */
+public class HierarchicalConfigService extends AbstractService {
+
+    public HierarchicalConfigService(RestTemplate restTemplate, GoodDataSettings settings) {
+        super(restTemplate, settings);
+    }
+
+    /**
+     * Returns aggregated project settings for given project.
+     *
+     * @param project project, cannot be null
+     * @return settings for given project
+     */
+    public Settings projectSettings(final Project project) {
+        notNull(project, "project");
+        try {
+            Settings settings = restTemplate.getForObject(PROJECT_SETTINGS_TEMPLATE.expand(project.getId()), Settings.class);
+
+            if (settings == null) {
+                throw new GoodDataException("empty response from API call");
+            }
+            return settings;
+        } catch (GoodDataException | RestClientException e) {
+            throw new GoodDataException("Unable to list project settings for project ID=" + project.getId(), e);
+        }
+    }
+
+    /**
+     * Returns setting for given project.
+     *
+     * @param project project, cannot be null
+     * @param settingName unique name (key) of setting item, cannot be empty
+     * @return setting for given project with given name (key)
+     */
+    public SettingItem getProjectSetting(final Project project, final String settingName) {
+        notNull(project, "project");
+        notEmpty(settingName, "settingName");
+
+        try {
+            String settingUri = PROJECT_SETTING_TEMPLATE.expand(project.getId(), settingName).toString();
+            SettingItem setting = restTemplate.getForObject(settingUri, SettingItem.class);
+
+            if (setting == null) {
+                throw new GoodDataException("Project setting cannot be retrieved from URI " + settingUri);
+            }
+            return setting;
+        } catch (GoodDataException | RestClientException e) {
+            throw new GoodDataException("Unable to get project setting: " + settingName, e);
+        }
+    }
+
+    /**
+     * Creates or update setting for given project.
+     *
+     * @param project project for which the setting item should be created/updated, cannot be null
+     * @param setting setting to be created/updated, cannot be null
+     * @return created/updated project setting
+     */
+    public SettingItem upsertProjectSetting(final Project project, final SettingItem setting) {
```

Comment:
I would rename it to `setProjectSetting`

Reply:
Changed